invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/index.js

```diff
@@ -78,78 +71,63 @@ class DashboardTabs extends Component {
       overrideStore.add(componentId, component);
     }
 
-    this.panes = this.ACTIVE_TAB_PANES.map((pane, index) => {
+    this.panels = TAB_PANELS.map((pane, index) => {
       const { appId, ...config } = getConfigFromDataAttribute(
         rootElement,
         _camelCase(pane.configDataAttribute)
       );
       return {
-        menuItem: (
-          <Menu.Item key={index} className="selected-menu-tab">
-            {pane.label}
-          </Menu.Item>
-        ),
-        render: () => (
-          <Container>
-            <Tab.Pane>
-              <SearchApp appName={appId} key={appId} config={config} />
-            </Tab.Pane>
-          </Container>
-        ),
+        menuLabel: pane.label,
+        appId: appId,
+        config: config,
       };
     });
   }
 
-  onTabChange = (e, data) => {
-    const activePane = this.ACTIVE_TAB_PANES[data.activeIndex];
-    replaceURLPathname(activePane.pathname);
+  handleItemClick = (e, { name }) => {
+    const defaultActiveMenuElement = this.routes.indexOf(_lowerCase(name));
+    this.setState({
+      defaultActiveMenuElement: defaultActiveMenuElement,
+    });
+    replaceURLPathname(_lowerCase(name));
   };
 
-
-
   render() {
-    const Element = Tab;
-    // apply custom menu rendering
-    Element.prototype.renderMenu = renderMenuCustom;
-
+    const { defaultActiveMenuElement } = this.state;
     return (
       <Container fluid>
-        <Element
-          id="dashboard-tab"
-          defaultActiveIndex={this.state.defaultActiveTab}
-          panes={this.panes}
-          menu={{ secondary: true, pointing: true }}
-          onTabChange={this.onTabChange}
-          renderActiveOnly={true}
-        />
+        <div id="dashboard-menu">
+          <Container fluid id="dashboard-menu-container">
+            <Container>
+              <Menu pointing secondary>
+                {this.panels.map(({ menuLabel }, index) => {
```

Comment:
I have designed an alternative solution that reduces the loops to one, but it may be a bit complex if readability is serious concern.

```
    const { currentActiveMenuElement } = this.state;
    const content = this.panels.reduce(
      (previousValue, { menuLabel, appId, config, key }) => {
        previousValue.menuItems.push(
          <Menu.Item
            name={menuLabel}
            key={key}
            active={key === currentActiveMenuElement}
            onClick={this.handleItemClick}
          />
        );
        key === currentActiveMenuElement &&
          (previousValue.view = (
            <Container key={key}>
              <Segment attached="bottom">
                <SearchApp appName={appId} key={appId} config={config} />
              </Segment>
            </Container>
          ));
        return previousValue;
      },
      { menuItems: [], view: null }
    );

    return (
      <Container id="dashboard-menu" fluid>
        <Container fluid id="dashboard-menu-container">
          <Container>
            <Menu pointing secondary>
              {content.menuItems}
            </Menu>
          </Container>
        </Container>
        {content.view}
      </Container>
    );
  }
  ```

Reply:
IMHO: Ideally the code generating the <Menu.Item> (the loop) should be moved to when the component is mounted (as it is the same always) and the render method should only take care of checking the active method.

Reply:
I'm not 100% sure of this:
To control if Menu.Item element is or isn't active, you must place the active property in the element itself. If we define the Menu.Item in the constructor, the condition will always remain the same (the component is not re-created, is only re-rendered, so the active condition must be placed in the render). Currently, when you press any menu item, the state changes, so the component is re-rendered, with a new active Menu.Item.